tools/shared.py

```diff
@@ -1638,10 +1638,12 @@ def has_substr(array, substr):
     if not has_substr(args, '-DCMAKE_TOOLCHAIN_FILE'):
       args.append('-DCMAKE_TOOLCHAIN_FILE=' + path_from_root('cmake', 'Modules', 'Platform', 'Emscripten.cmake'))
 
-    # On Windows specify MinGW Makefiles if we have MinGW and no other toolchain was specified, to avoid CMake
+    # On Windows specify MinGW Makefiles or ninja if we have them and no other toolchain was specified, to avoid CMake
     # pulling in a native Visual Studio, or Unix Makefiles.
     if WINDOWS and '-G' not in args and Building.which('mingw32-make'):
       args += ['-G', 'MinGW Makefiles']
+    if WINDOWS and '-G' not in args and Building.which('ninja'):
```

Comment:
this could be `elif`? or is it intentionally possibly passing in two options to cmake?

Reply:
oh, oops, you're right.
Arguably we should use Ninja in preference to mingw32-make too.

Does this code get used by every invocation of emcmake, or just the ones in tests?

Reply:
I believe it's called from `emconfigure`, so not just the test suite.

At least on my machine it looks like it just does the last `-G` command it receives. But I'm not sure if that's documented.

Reply:
OK, that's a bit odd because a user might be expecting default CMake behavior and not get it. but I'l keep the current default to avoid breaking anyone depending on that.